src/java.base/share/classes/java/lang/String.java

```diff
@@ -3126,13 +3126,50 @@ public String replace(CharSequence target, CharSequence replacement) {
      * Splits this string around matches of the given
      * <a href="../util/regex/Pattern.html#sum">regular expression</a>.
      *
-     * <p> The array returned by this method contains each substring of this
+     * <p> This method works as if by invoking the three-argument {@link
```

Comment:
The descriptions should be mostly self contained, repeating/duplicating the test as necessary so it easier to read. In this case, there is no reason to change the existing method javadoc except perhaps to note the relationship to the new method. It will be obvious that the spec and behavior are not changing.